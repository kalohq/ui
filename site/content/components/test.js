<h1 id="design-tokens">Design Tokens</h1>
<blockquote>
<p><strong>Note</strong>: The files in this directory are autogenerated. See <a href="/config/design-tokens">config/design-tokens</a> for the source.</p>
</blockquote>
