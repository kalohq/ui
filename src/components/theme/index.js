/* @flow */
export {default} from './theme';
