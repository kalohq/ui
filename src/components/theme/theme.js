/* @flow */

import tokens from '../../design-tokens/tokens.theme.js';

/**
 * ThemeProvider Kalo Theme
 * 
 * This is the core CSS-in-JS theme that is passed in to the
 * global ThemeProvider. It is mainly composed of our core
 * design tokens.
 */

const theme = {
  space: [0, 8, 16, 32, 64],
  breakpoints: [],
  ...tokens,
};

export default theme;
