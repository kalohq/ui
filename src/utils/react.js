import ReactDOM from 'react-dom';

/**
 * Focus a ref’d react input when it is mounted
 * @param ref
 */
export function focusOnMount(ref) {
  const el = ReactDOM.findDOMNode(ref);

  if (el) {
    // Give focus and put the cursor at the end of
    // the value.
    el.focus();
    el.value = el.value;
  }
}

/**
 * Select the text of a ref’d react input when it is mounted
 * @param ref
 */
export function selectOnMount(ref) {
  if (ref && ref.select) {
    ref.select();
  }
}
