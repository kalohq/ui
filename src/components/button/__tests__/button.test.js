/* @flow */
/* eslint-env jest */
import * as React from 'react';
import * as emotion from 'emotion';
import {createSerializer} from 'jest-emotion';
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'emotion-theming';
import sinon from 'sinon';

import {shallow, mount} from '../../../utils/test/enzyme';

import theme from 'components/theme';
import Button from '../button';

import reactStyles from '../button.react.css';

expect.addSnapshotSerializer(createSerializer(emotion));

describe('Button', () => {
  const defaultProps = {
    variant: 'tertiary',
    size: 'medium',
  };

  const create = props => (
    <Button {...defaultProps} {...props}>
      A Button
    </Button>
  );

  test('should render shallow component ok', () => {
    const element = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Button variant="tertiary" size="medium">
            A tertiary button
          </Button>
        </ThemeProvider>
      )
      .toJSON();
    expect(element).toMatchSnapshot();
  });

  test('should render with a success message', () => {
    const element = create({
      message: 'Hello from the success message',
    });

    const result = shallow(element);
    expect(
      result
        .dive()
        .find({className: reactStyles['ui-btn__react-success-message']}).length
    ).toBe(1);
  });

  test('should display with a loadingSpinner', () => {
    const element = create({
      loading: true,
    });

    const result = mount(element);

    expect(result.find('.ui-btn--loading').hostNodes().length).toBe(1);
  });

  test('should call a given onClick event on click', () => {
    const onButtonClick = sinon.spy();

    const element = create({
      onClick: onButtonClick,
    });

    const result = shallow(element);

    result.simulate('click', {
      preventDefault: () => {},
    });

    expect(onButtonClick.calledOnce).toBe(true);
  });

  test('should not allow onClick events to be set if disabled', () => {
    const onButtonClick = sinon.spy();

    const element = create({
      onClick: onButtonClick,
      disabled: true,
    });

    const result = shallow(element);

    result.dive().simulate('click', {
      preventDefault: () => {},
    });

    expect(onButtonClick.notCalled).toBe(true);
  });
});
