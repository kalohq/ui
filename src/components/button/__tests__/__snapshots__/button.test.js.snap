// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render shallow component ok 1`] = `
<button
  className="ui-btn ui-btn--medium ui-btn--tertiary emotion-0 emotion-1"
  data-test="ui-button"
  disabled={false}
  name={undefined}
  onClick={[Function]}
  role="button"
  style={Object {}}
  type={undefined}
>
  <div
    className="ui-btn__react-placeholder-message"
  >
    A tertiary button
  </div>
  <div
    className="ui-btn__react-visible-message"
    name="message"
    title="A tertiary button"
  >
    A tertiary button
  </div>
  <div
    className="ui-btn__react-success-message"
  />
</button>
`;
